plugins/modules/katello_activation_key.py

```diff
@@ -144,88 +114,144 @@ def override_to_boolnone(override):
-                            activation_key_entity.content_override(data={'content_override': {'content_label': label,
-                                                                         'value': str(override).lower()}})
+                }
+                desired_content_overrides = {
+                    product['label']: override_to_boolnone(product['override']) for product in content_overrides
+                }
+                changed_content_overrides = []
+
+                for label, override in desired_content_overrides.items():
+                    if override != current_content_overrides.pop(label, None):
+                        changed_content_overrides.append({'content_label': label, 'value': override})
+                for label in current_content_overrides.keys():
+                    changed_content_overrides.append({'content_label': label, 'reset': True})
+
+                if changed_content_overrides:
+                    payload = {
+                        'id': activation_key['id'],
+                        'content_overrides': changed_content_overrides,
+                    }
+                    module.resource_action('activation_keys', 'content_override', payload)
+                    changed = True
+
+            if host_collections is not None:
+                if 'host_collection_ids' in activation_key:
+                    current_host_collection_ids = set(activation_key['host_collection_ids'])
+                else:
+                    current_host_collection_ids = set(item['id'] for item in activation_key['host_collections'])
+                desired_host_collections = module.find_resources_by_name('host_collections', host_collections, params=scope, thin=True)
+                desired_host_collection_ids = set(item['id'] for item in desired_host_collections)
+
+                module.warn(repr((desired_host_collection_ids, current_host_collection_ids)))
+                if desired_host_collection_ids != current_host_collection_ids:
+                    ids_to_remove = current_host_collection_ids - desired_host_collection_ids
+                    if ids_to_remove:
+                        payload = {
+                            'id': activation_key['id'],
+                            'host_collection_ids': list(ids_to_remove),
+                        }
+                        module.resource_action('activation_keys', 'remove_host_collections', payload)
+                    ids_to_add = desired_host_collection_ids - current_host_collection_ids
+                    if ids_to_add:
+                        payload = {
+                            'id': activation_key['id'],
+                            'host_collection_ids': list(ids_to_add),
+                        }
+                        module.resource_action('activation_keys', 'add_host_collections', payload)
                     changed = True
 
         module.exit_json(changed=changed)
     except Exception as e:
```

Comment:
what kind of Exception can be raised here that is not yet caught by our `e2fail` decorator?

Reply:
i guess none.